File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    s = s.split('');
    var cnt = 0
    
    for ( let i = 0 ; i < s.length ; i++ ) {
        if ( s[i] === ')' ) {
            // 닫힐 괄호가 없는데 닫으려 하면 false로 변경 후 반복문 탈출
            if ( cnt === 0 ) {
                answer = false;                 
                break;
            } 
            else cnt -= 1;
        }
        // 여는 괄호면 cnt +1
        else cnt += 1;    
    }
    
    // 괄호가 완전히 닫히지 않았으면 false 반환
    if ( cnt !== 0 ) answer = false;

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cs
using System;
using System.Linq;

public class Solution {
    public bool solution(string s) {
        bool answer = true;
        int cnt = 0;
        var temp = s.Select(letter => letter).ToList();
        
        for (int i = 0; i < temp.Count; i++) {
            if (temp[i] == ')' && cnt == 0) {
                answer = false;
                break;
            }
            else if (temp[i] == ')' && cnt != 0) {
                cnt--;
            }
            else {
                cnt++;
            }
        }
        
        if (cnt != 0) {
            answer = false;
        }
        
        return answer;
    }
}